### ISSUE TITLE
Inconsistencies generating `plump` objects from HTML and JSON

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2533]
I observed that the following does not generate equivalent `plump` objects.

```
(plump:parse (ffi-buffer-get-document (current-buffer)))
```

```
(nyxt/dom::named-json-parse (nyxt/dom::get-document-body-json))
```

Here's a concrete example. Take `https://motherfuckingwebsite.com/` as the URL and generate the objects as above. Notice the discrepancies on the bottom of the following screenshot (JSON on the left, HTML on the right). In  the HTML version, `script` elements belong to the `fulltext-element` class, whereas in the JSON version it does not.

![2022-09-01_16:54:02](https://user-images.githubusercontent.com/45483512/187940330-77a9fa60-8d77-4e85-a380-8696e342d9ea.png)

I've already discussed the subject briefly with @aartaka, and he helped me to gain more insight.

To solve this we need to answer the following question first:

What the difference between feeding `plump:parser` with `ffi-buffer-get-document` and `(nyxt/dom::named-json-parse (nyxt/dom::get-document-body-json))`, if any? 

If there aren't, we might not need JSON after all.

If we do need JSON, then we need to ensure that the `plump` object we generate follows the same rules as the HTML parser uses (namely with respect to `fulltext-element`s).

**Comment by Ambrevar** [comment]
> Are the differences between feeding `plump:parser` with `ffi-buffer-get-document` and `(nyxt/dom::named-json-parse (nyxt/dom::get-document-body-json))`?

There is a phrasing mistake, what do you mean?

I'm not too familiar with this part of the code.  What was the intent of the `nyxt/dom:*json*` in the first place?

**Comment by aadcg** [comment]
> There is a phrasing mistake, what do you mean?

Sorry, fixed.

> I'm not too familiar with this part of the code. What was the intent of the nyxt/dom:*json* in the first place?

The goal is to manipulate the DOM from CL, instead of doing it via JS. 

What we're doing now is to convert the HTML structure into JSON with JS, and feeding the latter into `plump`. But that means that we actually re-implement lots of things on our side, namely `dom.lisp`, which is basically an interface so that `plump` parses JSON. 

I don't see how JSON actually differs (content-wise), since it that computation is based on the same HTML source.

I think things will become clearer when I'll draft a PR.

**Comment by aadcg** [comment]
For context, I'm working on #2529.

**Comment by aadcg** [comment]
This has been solved in  b25f4ab.
